rline/rline.go

```diff
@@ -123,9 +125,45 @@ func (l *Rline) Password(prompt string) (string, error) {
 	return "", ErrPasswordNotAvailable
 }
 
+// filterWriter implements io.Writer and applies a string filter.
+type filterWriter struct {
+	writer io.Writer
+	filter func(string) string
+}
+
+func (fw *filterWriter) Write(p []byte) (n int, err error) {
+	if fw.filter == nil {
+		return fw.writer.Write(p)
+	}
+	filtered := fw.filter(string(p))
+	return fw.writer.Write([]byte(filtered))
+}
+
+// Close implements io.Closer if the underlying writer is an io.Closer.
+func (fw *filterWriter) Close() error {
+	if closer, ok := fw.writer.(io.Closer); ok {
+		return closer.Close()
+	}
+	return nil
+}
+
 // SetOutput sets the output format func.
 func (l *Rline) SetOutput(f func(string) string) {
-	l.Inst.Config.Output = f
+	cfg := l.Inst.Config.Clone() // Get a mutable copy of the config
+
+	var currentWriter io.WriteCloser
+	if f != nil {
+		// Wrap the original stdout with the filter
+		currentWriter = &filterWriter{writer: l.originalStdout, filter: f}
+	} else {
+		// If filter is nil, revert to the original stdout
+		currentWriter = l.originalStdout
+	}
```

Comment:
I tend to write this this way


```suggestion
	var currentWriter io.WriteCloser

	// use original stdout by default
	currentWriter = l.originalStdout
	if f != nil {
		// Wrap the original stdout with the filter
		currentWriter = &filterWriter{writer: l.originalStdout, filter: f}
	}
```


What to you think?